pallets/stablewap/src/benchmarks.rs

```diff
@@ -0,0 +1,227 @@
+// This file is part of Basilisk-node
+
+// Copyright (C) 2020-2022  Intergalactic, Limited (GIB).
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use super::*;
+
+use frame_benchmarking::account;
+use frame_benchmarking::benchmarks;
+use frame_support::pallet_prelude::DispatchError;
+use frame_system::RawOrigin;
+use orml_traits::MultiCurrency;
+use orml_traits::MultiCurrencyExtended;
+use sp_runtime::Permill;
+
+use hydradx_traits::Registry;
+
+use crate::types::{Balance, PoolId};
+
+benchmarks! {
+	 where_clause {  where T::AssetId: From<u32> + Into<u32>,
+		T::Currency: MultiCurrencyExtended<T::AccountId, Amount=i128>,
+		T::AssetRegistry: Registry<T::AssetId, Vec<u8>, Balance, DispatchError>,
+		T: crate::pallet::Config
+	}
+
+	create_pool{
+		let token_a = T::AssetRegistry::create_asset(&b"one".to_vec(), 1u128)?;
+		let token_b = T::AssetRegistry::create_asset(&b"two".to_vec(), 1u128)?;
+
+		// Note: this is extreme case where calculate_d does around 40 iterations.
+		let initial_liquidity = (1_000_000_000_000_000_000_000u128, 10u128);
+
+		let amplification = 100u128;
```

Comment:
Does this matter in terms of benchmarks? If yes, is this the worst case?

Reply:
Current scenario to benchmarks does 37 iterations and that's quite extreme case, where reserves are really extreme. 

with this amp set to 100 - it performs 37 iterations.

And I just tried to set it to 1000 and it does less iterations.  Setting it to 5 instead - it actually performs 2 more iterations.

however, i believe this scenarios is good enough. From research and testing - we could see that on average - it converges within 7-10 iterations ( with "normal" values). 

We could not find any scenarios where it never converges or does significantly more iterations that 40.

Reply:
But we should either lower the number of iterations, because we can benchmark it as it is.

Or we can actually force the benchmark to do the max number of iterations to make sure that we cover the worst case for sure.